### ISSUE TITLE
empty-line-between-blocks should have an option to ignore single line blocks

### ISSUE THREAD

**Issue by BPScott** [sasstools/sass-lint#282]
Sometimes when writing many short blocks where there is a single property (/ mixin / extend) per block it is useful to group them close to each other as they are often inherently related. For instance a set of width classes can be densely packed instead of having a blank line between each class:

```scss
.w1-of-12 { width: (1/12) * 100%; }
.w2-of-12 { width: (2/12) * 100%; }
.w3-of-12 { width: (3/12) * 100%; }
.w4-of-12 { width: (4/12) * 100%; }
.w5-of-12 { width: (5/12) * 100%; }
.w6-of-12 { width: (6/12) * 100%; }
.w7-of-12 { width: (7/12) * 100%; }
.w8-of-12 { width: (8/12) * 100%; }
.w9-of-12 { width: (9/12) * 100%; }
.w10-of-12 { width: (10/12) * 100%; }
.w11-of-12 { width: (11/12) * 100%; }
.w12-of-12 { width: (12/12) * 100%; }
```

[Scss-lint's empty-line-between-blocks rule](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#emptylinebetweenblocks) allows these single-line rule sets to sit next to each other as it has a configuration option `allow_single_line_rule_sets` that defaults to `true`. The configuration option may be set to off to disallow this exception.

Sass-lint should mimic this behaviour so that I may write the above without raising warnings.

**Comment by DanPurdy** [comment]
I'm good with this, makes a lot of sense.

**Comment by bgriffith** [comment]
Yep agree with this.

How about `ignore-single-line-rulesets` as the option? Since I feel you're instructing the rule to ignore them from potentially being followed by a newline.

**Comment by bgriffith** [comment]
On second though, I like the idea of using `allow-single-line-rulesets` as the option name for all similar rules. Thoughts?

**Comment by BPScott** [comment]
No strong opinion about if we go with `ignore-single-line-rulesets` over `allow-single-line-rule-sets`, (I'm just the guy coming up with requests and doing no work :)) though think that rules with this sort of option should be either always positive or always negative across the whole project for the sake of consistency, i.e. all options that are this sort of boolean toggle should be `allow-thing`or `ignore-thing`.

Having a quick flick through the docs, [no-qualifying-elements](https://github.com/sasstools/sass-lint/blob/develop/docs/rules/no-qualifying-elements.md) has `allow-element-with-attribute`, `allow-element-with-class` and `allow-element-with-id` but that seems to be the only piece of prior-art. 

Personally I think naming these toggle options as positive terms (e.g. allow-thing, enable-thing) is clearer as `allow: true` is easier to parse than the double negative of `ignore: false` despite both having the same effect.

**Comment by DanPurdy** [comment]
Agreed. We have made many rules fully negative in the sense they are named `no-*` so any exceptions to our rules should be positive. I think consistently having `allow` also makes it clearer for people scanning through the config file.

Nice catch, I'll put it to @bgriffith and possibly add some documentation on this going forward.
